src/server/scripts/EasternKingdoms/zone_elwynn_forest.cpp

```diff
@@ -275,7 +275,260 @@ struct npc_cameron : public ScriptedAI
     GuidVector _childrenGUIDs;
 };
 
+/*######
+## npc_supervisor_raelen
+######*/
+
+enum SupervisorRaelen
+{
+    EVENT_FIND_PEASANTS  = 1,
+    EVENT_NEXT_PEASANT   = 2,
+    NPC_EASTVALE_PEASANT = 11328
+};
+
+struct npc_supervisor_raelen : public ScriptedAI
+{
+    npc_supervisor_raelen(Creature* creature) : ScriptedAI(creature) {}
+
+    void Reset() override
+    {
+        _PeasantId = 0;
+        peasantGUIDs.clear();
+        _events.ScheduleEvent(EVENT_FIND_PEASANTS, 4s);
+    }
+
+    void SetData(uint32 /*type*/, uint32 data) override
+    {
+        if (data == 1)
+        {
+            ++_PeasantId;
+            if (_PeasantId == 5) _PeasantId = 0;
+            _events.ScheduleEvent(EVENT_NEXT_PEASANT, 2s, 6s);
+        }
+    }
+
+    void CallPeasent()
+    {
+        if (Creature* peasant = ObjectAccessor::GetCreature(*me, peasantGUIDs[_PeasantId]))
+            peasant->AI()->SetData(1, 1);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        if (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_FIND_PEASANTS:
+                {
+                    GuidVector tempGUIDs;
+                    std::list<Creature*> peasants;
+                    GetCreatureListWithEntryInGrid(peasants, me, NPC_EASTVALE_PEASANT, 100.f);
+                    for (Creature* peasant : peasants)
+                    {
+                        tempGUIDs.push_back(peasant->GetGUID());
+                    }
+                    peasantGUIDs.push_back(tempGUIDs[2]);
+                    peasantGUIDs.push_back(tempGUIDs[3]);
+                    peasantGUIDs.push_back(tempGUIDs[0]);
+                    peasantGUIDs.push_back(tempGUIDs[1]);
+                    peasantGUIDs.push_back(tempGUIDs[4]);
+                    _events.ScheduleEvent(EVENT_NEXT_PEASANT, 1s);
+                    break;
+                }
+                case EVENT_NEXT_PEASANT:
+                    CallPeasent();
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+private:
+    EventMap _events;
+    uint8 _PeasantId;
```

Comment:
Please initialize this to 0 (even if it's done in Reset() also)

Reply:
done